Design/ThirdpartyDependencies.md

```diff
@@ -0,0 +1,300 @@
+* Homegrown set of scripts are very clunky.
+* Maintenance of build scripts adds overhead.
+* Complete bundling of dependencies gives a much larger user installer package.
+
+
+### MacOS
+
+MacOS uses the [Homebrew](https://brew.sh/) package manager along with a custom
+set of [formulae](https://github.com/mantidproject/homebrew-mantid) for obselete
+dependencies.
+Homebrew either downloads a prebuilt binary or builds it locally from the formula.
+A developer formula is provided to easily install the required dependencies.
+
+The package provided for users contains all required dependencies
+much like Windows.
+
+Pros:
+* Build formula (mostly) maintained for us.
+* Simply commands to install packages
+
+Cons:
+* Dependencies not versioned with the main code. Hard to track versions used.
+* No way to set deployment. We have to build on the earliest OS we intend
+  to support causing issues with hardware availability.
+* New dependencies have to be installed by hand on developers
+  and builders machines.
+* Versions can be pinned but Homebrew really likes to have everything at HEAD
+
+## General Concerns
+
+While each approach has the drawbacks described above
+there are also general issues with the current approach:
+
+1. It is not possible to define a version of a given
+   library as the version used across the whole project.
+   This can cause confusion for developers
+   where their platform might have a newer api
+   yet their code does not work on other platforms.
+   This slows down development.
+2. Using a new library takes a lot of overhead and
+   in some cases is basically impossible
+3. Three different methods requires much additional overhead
+   in understanding how things are managed.
+
+## Requirements
+
+The following is a list of requirements that must be met by any solution:
+
+1. Supports C++ & Python packages
+2. allow mantid to select any version of a given dependency
+3. version definitions alongside the code in the repository
+4. use the same version of a dependency across all platforms
+5. provide users a single "package" to install self-contained Workbench
+   application along with all required libraries. 
+6. must be able to install versions side-by-side
+   including nightly and unstable versions.
+
+## Use Cases
+
+1. Ship a self-contained desktop application
+   with the latest versions of required libraries,
+   also allowing users to install their own Python packages
+2. Provide a `mantid` framework library for users to use in
+   their own projects, e.g mslice, PyRS + other SNS GUIs
+
+## Solutions
+
+Here we describe possible solutions to the problems outlined above.
+The solutions considered in detail are:
+
+* [Conan](#Conan)
+* [Conda](#Conda)
+
+Other solutions not considered:
+
+* Flatpak/Snap - would only work for Linux and still be a large effort.
+* Singularity/Docker - unsure about desktop applications & would not work for macOS.
```

Comment:
(Docker on Windows requires the Hyper-V role last time I checked, so users would have to get admin access and have a processor capable of exposing the required instructions - so it's not universally compatible)